### ISSUE TITLE
EF Core IdentityDbContext UserStore Issue when not deriving from IdentityDbContext

### ISSUE THREAD

**Issue by dazinator** [dotnet/aspnetcore#12511]
## Describe the bug
I have created a repro here to demonstrate the issue, just run and step through the console app: https://github.com/dazinator/EfIdentity5793Repro

The issue occurs in the following scenario:

1. You want to use EF Core identity, but you can't derive your applications DbContext class from `IdentityDbContext<>` (because you need to derive from an alternative) - however you still wish to use asp.net core Identity backed by EF so you look to add the EF core entities and configuration for identity to your own DbContext instance

in conjunction with:

2. 

```
 services.AddIdentityServices<User, Role>()
                       .AddEntityFrameworkStores<MyDbContext>()

```

The `AddEntityFrameworkStores` adds a UserStore implementation that doesn't correctly detect the custom type of IdentityUserClaim<TKey> and others. This can be seen here where it silently skips over this case, leading to runtime errors later on: https://github.com/aspnet/AspNetCore/blob/bfec2c14be1e65f7dd361a43950d4c848ad0cd35/src/Identity/EntityFrameworkCore/src/IdentityEntityFrameworkBuilderExtensions.cs#L54

The runtime error can be seen, if for example you attempt to use `UserManager<>` some methods that work purely with Users and Roles will work, but when it comes to Claims, you get exceptions saying `IdentityUserClaim<TKey>` hasn't been configured on the EF model.

### To Reproduce
1. Run this console app: https://github.com/dazinator/EfIdentity5793Repro

### Expected behavior
Configuration should throw an error early, and guidance should be offered in how to configure custom derived types for IdentityUserClaim<t> when not using a derived IdentityDbContext.
### Additional context
Include the output of `dotnet --info`

**Comment by blowdart** [comment]
@ajcvickers Thoughts?

**Comment by ajcvickers** [comment]
@blowdart Seems like a not unreasonable, but not super high priority, request for guidance and docs.  We're certainly don't have time to do anything here for 3.0.

**Comment by dazinator** [comment]
Irrespective of guidance or docs, my issue is more specifically with this code here: https://github.com/aspnet/AspNetCore/blob/bfec2c14be1e65f7dd361a43950d4c848ad0cd35/src/Identity/EntityFrameworkCore/src/IdentityEntityFrameworkBuilderExtensions.cs#L54 

I'm suggesting it's a bug to silently skip over and not set up stores correctly for Claims etc, when the DbContext doesnt derive from IdentityDbContext. If you are saying this is not a bug, then it seems you have to configure your own stores when using a custom DbContext - but in that case AddEntiyyFrameworkStores() is misleading as it's not clear it only set up some and not others.

**Comment by ajcvickers** [comment]
@dazinator Agreed that it is somewhat misleading.

**Comment by dazinator** [comment]
If entity configuration was put into `IEntityTypeConfiguration<T>` implementations this would make using identity a lot simpler when you aren't driving your dbcontext from IdentityDbContext.

**Comment by dazinator** [comment]
PR Submitted to move entity configuration into EF Core entity configuration classes.

**Comment by urielzen** [comment]
I am facing the "Cannot create a DbSet for 'IdentityUserClaim<string>' because this type is not included in the model for the context" exception.

This happened because I did not have any use for the role related tables, so I changed from:
public class ApplicationDbContext : IdentityDbContext<ApplicationUser>

to:
public class ApplicationDbContext : IdentityUserContext<ApplicationUser, string, ApplicationUserClaim, ApplicationUserLogin, ApplicationUserToken>

Could anyone confirm, if this fix will also address the issue I'm encountering?

I appreciate it!
